Builder Application Proccess - Non-builders can apply to join a cohort

Avelous opened the pull request:
cohort/0x123.../apply page allows a non builder apply to join a cohort and 

cohort/0x123.../applications allows an admin to manage (approve or reject) those applications. 

### Steps to test 

**Builder**
- Connect a non builder wallet to an existing cohort and click the "apply to be a builder" button
- Navigate to the apply page, click apply and a modal pops up, fill in the fields and submit

**Admin**
- Connect an admin to the cohort and click the applications button which displays only when there are pending applications
- Navigate to the applications page to view all applications and either approve or reject pending applications


### Pr also fixes issue #62 

<img width="1795" alt="Screenshot at May 06 16-45-37" src="https://github.com/user-attachments/assets/7571ba91-712f-40ff-a24d-d4dabbf96983" />


<img width="1799" alt="Screenshot at May 06 16-45-18" src="https://github.com/user-attachments/assets/041c56c0-e986-4395-8c8b-ef5d1da9ab86" />
<img width="1799" alt="Screenshot at May 06 16-45-28" src="https://github.com/user-attachments/assets/f00b8d9c-7f75-4cde-8132-76bb382e9b94" />


<img width="1799" alt="Screenshot at May 06 16-43-50" src="https://github.com/user-attachments/assets/39c7a1bb-28fb-4c9f-86ca-4c672bc36725" />

ZakGriffith reviewed (commented):
Process all worked smoothly for me.

The applications page for the admins is good, let's add a button to get to it from the main page as well, along with the button that currently exists in the members page.  

The builder process could use a little tweaking.  If they have not yet applied, there could just be the apply button which opens the modal to apply(and takes them to their application page after they submit).  If they have already applied, there would be a view application button displayed which takes them to the application page to see their open application.  

On the builder's view of the application page, it's a little busy with text.  We can remove the "Apply to enter cohort" and the Applications rectangle just leaving Your Applications.
<img width="402" alt="image" src="https://github.com/user-attachments/assets/83a246bc-3b2c-4dca-9ee3-880ad0b33217" />

Let's add some filtering in the github text inputs so they can't add spaces or non-allowed characters on accident.  Looks like github allows only alphanumeric characters and dashes in the usernames and a max of 30 characters.

Other than those tweaks, it's all working well!

ZakGriffith commented:
Forgot to mention, let's also add a selector when deploying a cohort to turn on or off the ability for any random builder to apply.

Avelous commented:
Pused commits that

- Moved builder application modal button to the cohort main page
- Cleaned up builder applicacions screen of redundant texts 
- Restricted github usernames to alphanumeric characters
- Added admin setting to allow builder applications to join cohort

@ZakGriffith

ZakGriffith commented:
Excellent, this has a real good feel to it!  One final tweak before merge would be a 24 hour notification when they get rejected, could be placed to the right of the Apply To Join button.

![image](https://github.com/user-attachments/assets/0aefee3f-edcf-4a7a-9f7a-1031cb62b504)

Avelous commented:
Pushed a commit for applicant rejection notifications

ZakGriffith commented:
I see the notification here, but it feels like there should be a rejection message that goes along with it.  When I click the button or the notification, I feel it should tell me I was rejected, but it bring up the apply modal.  Creating a rejection message medal would explain to the builder that they were rejected.  Another good move would be to move the notification to be next to the button instead of on top (with it's own icon the same way the withdraw requests look).  That way they can skip right to apply again, or click the notification icon and see a rejection modal.

![image](https://github.com/user-attachments/assets/4a7ca41c-895e-45d8-babe-c392534d74ed)

This bring up another idea. When an admin is rejecting, they can type in a short rejection message.  That message would them be displayed in the rejection modal, and also displayed on the Manage Applications page for admin.  Thoughts?

ZakGriffith commented:
Looking good, merging
